src/libprojectM/Renderer/Renderer.cpp

```diff
@@ -988,22 +841,29 @@ void Renderer::draw_stats()
 	stats += "Preset:""\n";
 	stats += "Warp Shader: " + warpShader + "\n";
 	stats += "Composite Shader: " + compShader + "\n";
-	drawText(stats.c_str(), 30, 20, 2.5);
+    m_menuText.DrawBegin();
+	m_menuText.Draw(stats, 30, 20, 2.5);
+    m_menuText.DrawEnd();
 #endif /** USE_TEXT_MENU */
 }
 
 // TODO
 void Renderer::draw_fps()
 {
 #ifdef USE_TEXT_MENU
-	drawText(this->fps().c_str(), 30, 20, 2.5);
+    m_menuText.DrawBegin();
+	m_menuText.Draw(this->fps(), 30, 20, 2.5);
+    m_menuText.DrawEnd();
 #endif /** USE_TEXT_MENU */
 }
 
 void Renderer::draw_toast()
 {
 #ifdef USE_TEXT_MENU
-	drawText(this->toastMessage().c_str(), (vw/2), (vh/2), 2.5, GLT_CENTER, GLT_CENTER);
+    m_menuText.DrawBegin();
```

Comment:
It could probably be a macro or wrapper that takes a lambda

Reply:
For a single line, sure. Or just a boolean parameter `singleLine` that calls begin/end automatically if true.

As discussed on Discord, this whole text rendering should prbably be outsourced into a separate component that injects drawables similar to the rest of the rendering pipeline, so the `Renderer` class doesn't need to care about this stuff. Saving that for later.